core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -181,24 +188,33 @@ static void configure(Environment environment, boolean filterBadDefaults) throws
     /**
      * Reads and returns the specified {@code policyFile}.
      * <p>
-     * Resources (e.g. jar files and directories) listed in {@code codebases} location
+     * Jar files listed in {@code codebases} location
      * will be provided to the policy file via a system property of the short name:
      * e.g. <code>${codebase.joda-convert-1.2.jar}</code> would map to full URL.
      */
     @SuppressForbidden(reason = "accesses fully qualified URLs to configure security")
-    static Policy readPolicy(URL policyFile, URL codebases[]) {
+    static Policy readPolicy(URL policyFile, Set<URL> codebases) {
         try {
             try {
                 // set codebase properties
                 for (URL url : codebases) {
                     String shortName = PathUtils.get(url.toURI()).getFileName().toString();
-                    System.setProperty("codebase." + shortName, url.toString());
+                    if (shortName.endsWith(".jar") == false) {
```

Comment:
There is absolutely nothing I can do here. Otherwise it will break IDEs running these shitty tests, where no sysprop is set :( Thats always been the problem with so much of this stuff!